pkg/lib/controller-runtime/client/ssa_test.go

```diff
@@ -0,0 +1,35 @@
+package client
+
+import (
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"testing"
+)
+
+func TestSetDefaultGroupVersionKind(t *testing.T) {
+	tests := []struct {
+		name string
+		s    *runtime.Scheme
+		obj  Object
+		want schema.GroupVersionKind
+	}{
+		{
+			name: "DefaultGVK",
+			s:    nil,
+			obj:  &corev1.ServiceAccount{},
+			want: schema.GroupVersionKind{
+				Group:   "",
+				Version: "v1",
+				Kind:    "ServiceAccount",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			SetDefaultGroupVersionKind(tt.obj, tt.s)
+			require.EqualValues(t, tt.obj.GetObjectKind().GroupVersionKind(), tt.want)
+		})
+	}
+}
```

Comment:
Having a unit test here that uses server side apply to mimic the same functionality that we want in the hot path (editing a pod label) via the fake client would be great. Not sure if we can SSA via the fake client. Would be nicer than another e2e test, but either approach works. I think the existing image polling e2e test covers the base case